mdtraj/core/topology.py

```diff
@@ -613,7 +651,7 @@ def from_dataframe(cls, atoms, bonds=None):
                     atom["resSeq"],
                     atom["segmentID"],
                 )
-
+            assert r is not None, "Expected a Residue, but got None"
```

Comment:
Is this a new check?

Reply:
Yes, this is to check the Residue (r) is not None before it's used, similar to the Chain is not None check on line 627